lib/connection.js

```diff
@@ -1372,6 +1372,14 @@ Connection.prototype.setClient = function setClient(client) {
   return this;
 };
 
+Connection.prototype.syncIndexes = async function syncIndexes() {
+  const result = [];
```

Comment:
Instead of returning an array, let's make this return an object where the keys are model names and the value is a result. So `result[model.modelName] = await this.model(model).syncIndexes()`

Reply:
@vkarpov15

Maybe is it better to return object, where key is `modelName` and value is result (successful) from `this.model(model).syncIndexes()` or if something went wrong instance of the Error (or MongooseError)?

So the result object will look like:

```js
const syncIndexesResult = await connection.syncIndexes();
// `syncIndexesResult`
{
  model1: ['index-1-1', 'index-1-2'], // this is example of the successful result from `Model.syncIndexes`,
  model2: MongooseError, 
  model3: [...],
}
```

As a result, you could control future actions (ex: show in console list of failed models or manually rerun syncIndexes on failed models).

What do you think?

Reply:
@vkarpov15 @AbdelrahmanHafez ping